driver/src/main/scala/api/commands/bson/instanceadministration.scala

```diff
@@ -54,30 +54,29 @@ object BSONCollStatsImplicits {
   implicit object CollStatsWriter extends BSONDocumentWriter[ResolvedCollectionCommand[CollStats]] {
     def write(command: ResolvedCollectionCommand[CollStats]): BSONDocument =
       BSONDocument(
-        "collStats" -> command.collection,
-        "scale" -> command.command.scale)
+        "collStats" -> command.collection, "scale" -> command.command.scale)
   }
+
   implicit object CollStatsResultReader extends DealingWithGenericCommandErrorsReader[CollStatsResult] {
-    def readResult(doc: BSONDocument): CollStatsResult =
-      CollStatsResult(
-        doc.getAs[String]("ns").get,
-        doc.getAs[Int]("count").get,
-        doc.getAs[BSONNumberLike]("size").get.toDouble,
-        doc.getAs[Double]("avgObjSize"),
-        doc.getAs[BSONNumberLike]("storageSize").get.toDouble,
-        doc.getAs[Int]("numExtents").get,
-        doc.getAs[Int]("nindexes").get,
-        doc.getAs[Int]("lastExtentSize").get,
-        doc.getAs[Double]("paddingFactor").get,
-        doc.getAs[Int]("systemFlags"),
-        doc.getAs[Int]("userFlags"),
-        doc.getAs[Int]("totalIndexSize").get,
-        {
-          val indexSizes = doc.getAs[BSONDocument]("indexSizes").get
+    def readResult(doc: BSONDocument): CollStatsResult = CollStatsResult(
+      doc.getAs[String]("ns").get,
+      doc.getAs[BSONNumberLike]("count").map(_.toInt).get,
```

Comment:
Get all numeric values as `BSONNumberLike`, to avoid type issues due to MongoDB internal cases.